thirdparty/patches/avro-1.10.2.c.patch

```diff
@@ -0,0 +1,84 @@
+--- CMakeLists.txt
++++ CMakeLists.txt
+@@ -18,7 +18,7 @@
+ #
+ cmake_minimum_required(VERSION 3.1)
+ project(AvroC C)
+-enable_testing()
++# enable_testing()
+
+ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} ${CMAKE_SOURCE_DIR})
+
+@@ -136,12 +136,14 @@ include_directories(${AvroC_SOURCE_DIR}/src)
+
+
+ # Enable codecs
+-
++set ( ZLIB_INCLUDE_DIR $ENV{TP_INSTALL_DIR}/include )
++set ( ZLIB_LIBRARIES $ENV{TP_INSTALL_DIR}/lib/libz.a )
+ find_package(ZLIB)
+ if (ZLIB_FOUND)
+     set(ZLIB_PKG zlib)
+     add_definitions(-DDEFLATE_CODEC)
+-    include_directories(${ZLIB_INCLUDE_DIRS})
++    include_directories(${ZLIB_INCLUDE_DIR})
++    link_directories(${ZLIB_LIBRARIES})
+     message("Enabled deflate codec")
+ else (ZLIB_FOUND)
+     set(ZLIB_PKG "")
+@@ -149,17 +151,17 @@ else (ZLIB_FOUND)
+     message("Disabled deflate codec. zlib not found.")
+ endif (ZLIB_FOUND)
+
+-find_package(Snappy)
+-if (SNAPPY_FOUND AND ZLIB_FOUND)  # Snappy borrows crc32 from zlib
+-    set(SNAPPY_PKG libsnappy)
+-    add_definitions(-DSNAPPY_CODEC)
+-    include_directories(${SNAPPY_INCLUDE_DIRS})
+-    message("Enabled snappy codec")
+-else (SNAPPY_FOUND AND ZLIB_FOUND)
+-    set(SNAPPY_PKG "")
+-    set(SNAPPY_LIBRARIES "")
+-    message("Disabled snappy codec. libsnappy not found or zlib not found.")
+-endif (SNAPPY_FOUND AND ZLIB_FOUND)
++# find_package(Snappy)
```

Comment:
why does these package need these patches?

Reply:
Two reasons
1.  Avro doesn't currently use snappy.
2. A compilation error occurs after this compilation item is enabled.